jupyterhub_moss/spawner.py

```diff
@@ -213,12 +214,20 @@ def options_from_form(self, formdata: Dict[str, List[str]]) -> Dict[str, str]:
                 raise RuntimeError("GPU(s) not available for this partition")
             options["gres"] = gpu_gres_template.format(options["ngpus"])
 
+        partition_environments = tuple(
+            self.partitions[partition]["jupyter_environments"].values()
+        )
         if "environment_path" not in options:
             # Set path to use from first environment for the current partition
-            default_venv = tuple(
-                self.partitions[partition]["jupyter_environments"].values()
-            )[0]
-            options["environment_path"] = default_venv["path"]
+            options["environment_path"] = partition_environments[0]["path"]
+
+        # Add environment_path to PATH unless it is in the settings with add_to_path=False
```

Comment:
I feel the comment should be more meaningful. We should explain the fact that custom envs are always added to the path and because of this, we must first check if the env is a default env.

And _then_, if the env is a default env, we check if it should be added to the path or not.

For example. I would rename `corresponding_env` in `corresponding_default_env` or `matching_default_env`.

And then above the condition `if matching_default_env is None`, I would add the comment `Custom env are always added to the PATH`